src/plugins/SectionCapsPlugin/SectionCapsPlugin.js

```diff
@@ -0,0 +1,362 @@
+            const vertex = vertices[key];
+            const index = indices[key];
+            geometries[key] = this.createGeometry(vertex, index);
+        }
+        return geometries;
+    }
+
+    createCSGPlane(sectionPlanes) {
+        const vertices = [
+            20, 20, 0.01,
+            -20, 20, 0.01,
+            -20, -20, 0.01,
+            20, -20, 0.01,
+            20, 20, 0.01,
+            20, -20, 0.01,
+            20, -20, -0.01,
+            20, 20, -0.01,
+            20, 20, 0.01,
+            20, 20, -0.01,
+            -20, 20, -0.01,
+            -20, 20, 0.01,
+            -20, 20, 0.01,
+            -20, 20, -0.01,
+            -20, -20, -0.01,
+            -20, -20, 0.01,
+            -20, -20, -0.01,
+            20, -20, -0.01,
+            20, -20, 0.01,
+            -20, -20, 0.01,
+            20, -20, -0.01,
+            -20, -20, -0.01,
+            -20, 20, -0.01,
+            20, 20, -0.01
+        ];
+
+        const indices = [
+            0, 1, 2, 0, 2, 3,            // front
+            4, 5, 6, 4, 6, 7,            // right
+            8, 9, 10, 8, 10, 11,         // top
+            12, 13, 14, 12, 14, 15,      // left
+            16, 17, 18, 16, 18, 19,      // bottom
+            20, 21, 22, 20, 22, 23
+        ]
+
+        let csgPlaneMerged = null;
+
+        sectionPlanes.forEach((sectionPlane) => {
+            const planeNormal = this.getObjectNormal(vertices, indices);
+            const normalizedDir = math.normalizeVec3(sectionPlane.dir);
+            const rotationQuaternion = this.computeQuaternionFromVectors(planeNormal, normalizedDir);
+
+            const planeTransformMatrix = this.getTransformationMatrix(sectionPlane.pos, rotationQuaternion, true);
+            const planeTransformedVertices = this.transformVertices(vertices, planeTransformMatrix);
+
+            const csgPlane = this.createGeometry(planeTransformedVertices, indices);
+
+            if (csgPlaneMerged)
+                csgPlaneMerged = csgPlaneMerged.union(csgPlane);
+            else
+                csgPlaneMerged = csgPlane;
+        })
+        console.log("csgPlaneMerged: ", csgPlaneMerged);
+        return csgPlaneMerged;
+    }
+
+    getCapGeometries(csgGeometries, csgPlane) {
+        const cappedGeometries = {};
+        for (const key in csgGeometries) {
+            const geometry = csgGeometries[key];
+            if (geometry.polygons.length) {
+                const intersected = geometry.intersect(csgPlane);
+                cappedGeometries[key] = intersected;
+            }
+
+        }
+
+        return cappedGeometries;
+    }
+
+    addIntersectedGeometries(csgGometries) {
+        for (const key in csgGometries) {
+            const webglGeometry = this.csgToWebGLGeometry(csgGometries[key]);
+            const model = this.addGeometryToScene(webglGeometry.vertices, webglGeometry.indices, key);
+            if (model)
+                this.prevIntersectionModels[key] = model;
```

Comment:
Let's put an underscore in front names of all methods and variables that are only used internally by the plugin, and ensure they're not in the TypeScript definition